src/Rule/Nette/RegularExpressionPatternRule.php

```diff
@@ -0,0 +1,110 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rule\Nette;
+
+use Nette\Utils\RegexpException;
+use Nette\Utils\Strings;
+use PhpParser\Node;
+use PhpParser\Node\Expr\StaticCall;
+use PHPStan\Analyser\Scope;
+use PHPStan\Rules\Rule;
+use PHPStan\Rules\RuleErrorBuilder;
+use PHPStan\Type\Constant\ConstantStringType;
+use PHPStan\Type\TypeUtils;
+use function in_array;
+use function sprintf;
+use function strtolower;
+
+/**
+ * @implements Rule<Node\Expr\FuncCall>
+ */
+class RegularExpressionPatternRule implements Rule
+{
+
+	public function getNodeType(): string
+	{
+		return StaticCall::class;
+	}
+
+	public function processNode(Node $node, Scope $scope): array
+	{
+		$patterns = $this->extractPatterns($node, $scope);
+
+		$errors = [];
+		foreach ($patterns as $pattern) {
+			$errorMessage = $this->validatePattern($pattern);
+			if ($errorMessage === null) {
+				continue;
+			}
+
+			$errors[] = RuleErrorBuilder::message(sprintf('Regex pattern is invalid: %s', $errorMessage))->build();
+		}
+
+		return $errors;
+	}
+
+	/**
+	 * @return string[]
+	 */
+	private function extractPatterns(StaticCall $staticCall, Scope $scope): array
+	{
+		if (!$staticCall->class instanceof Node\Name || !$staticCall->name instanceof Node\Identifier) {
+			return [];
+		}
+		$className = (string) $staticCall->class;
+		if ($className != Strings::class) {
+			return [];
+		}
+		$methodName = strtolower((string) $staticCall->name);
+		if (
+			!in_array($methodName, [
+				'split',
+				'match',
+				'matchall',
+				'replace',
+			], true)
+		) {
+			return [];
+		}
+
+		if (!isset($staticCall->getArgs()[1])) {
+			return [];
+		}
+		$patternNode = $staticCall->getArgs()[1]->value;
+		$patternType = $scope->getType($patternNode);
+
+		$patternStrings = [];
+
+		foreach (TypeUtils::getConstantStrings($patternType) as $constantStringType) {
+			$patternStrings[] = $constantStringType->getValue();
+		}
+
+		foreach ($patternType->getConstantArrays() as $constantArrayType) {
```

Comment:
I copied this from the rule in PHPStan but I'm wondering if `TypeUtils::getConstantArrays()` should be used here instead similar to `TypeUtils::getConstantStrings()` a few lines above. What's the difference?

Reply:
Failed tests say that `$patternType->getConstantArrays()` doesn't exist in older version. Should I increase the required PHPstan version to 1.9.0 where it was added or should I use `TypeUtils`?

Reply:
Ah just noticed that `TypeUtils::getConstantArrays()` is actually deprecated so I'll increase the required version then.